earthaccess/search.py

```diff
@@ -291,6 +291,23 @@ def debug(self, debug: bool = True) -> Self:
         self._debug = debug
         return self
 
+    def has_granules(self, has_granules: bool = True) -> Self:
+        """Only match collections with granules.
+
+        Parameters:
+            has_granules: If `True`, only return collections with granules. If
+            `False`, only return collections without granules.
+
+        Returns:
+            self
+        """
+        if not isinstance(has_granules, bool):
+            raise TypeError("has_granules must be of type bool")
+
+        self.params["has_granules"] = has_granules
+
+        return self
```

Comment:
Since has_granules is set to True by default when creating a new DataCollections instance, we should allow it to be "unset" by allowing users to pass `None`:

```suggestion
    def has_granules(self, has_granules: bool | None = True) -> Self:
        """Match only collections with granules, without granules, or either.

        Parameters:
            has_granules:
                If `True`, only return collections with granules. If
                `False`, only return collections without granules.
                If `None`, return both types of collections.

        Returns:
            self
        """
        if has_granules is not None and not isinstance(has_granules, bool):
            raise TypeError("has_granules must be of type bool or None")

        if has_granules is None and "has_granules" in self.params:
            del self.params["has_granules"]
        else:
            self.params["has_granules"] = has_granules

        return self
```

Reply:
:100: